appcues/src/main/java/com/appcues/ui/primitive/TextInputPrimitive.kt

```diff
@@ -60,6 +54,7 @@ internal fun TextInputPrimitive.Compose(modifier: Modifier) {
             onValueChange = {
                 if (maxLength == null || it.length <= maxLength) {
                     text.value = it
+                    formState.setValue(this@Compose, it)
```

Comment:
these primitives are mostly simplified now, but they do have this one additional requirement of reporting back to the persisted form state on update.